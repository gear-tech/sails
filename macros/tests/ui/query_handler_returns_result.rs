use sails_macros::query_handlers;

#[query_handlers]
mod queries {
    fn this(value: u32) -> u32 {
        value
    }
}

fn main() {}
