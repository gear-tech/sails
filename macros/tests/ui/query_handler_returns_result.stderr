error[E0308]: mismatched types
 --> tests/ui/query_handler_returns_result.rs:3:1
  |
3 | #[query_handlers]
  | ^^^^^^^^^^^^^^^^^
  | |
  | expected `Result<_, _>`, found `u32`
  | expected due to this
  |
  = note: expected enum `Result<_, _>`
             found type `u32`
  = note: this error originates in the attribute macro `query_handlers` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
 --> tests/ui/query_handler_returns_result.rs:3:1
  |
3 | #[query_handlers]
  | ^^^^^^^^^^^^^^^^^
  | |
  | expected `u32`, found `Result<_, _>`
  | arguments to this enum variant are incorrect
  |
  = note: expected type `u32`
             found enum `Result<_, _>`
note: tuple variant defined here
 --> tests/ui/query_handler_returns_result.rs:3:1
  |
3 | #[query_handlers]
  | ^^^^^^^^^^^^^^^^^
  = note: this error originates in the attribute macro `query_handlers` (in Nightly builds, run with -Z macro-backtrace for more info)
